src/main/java/de/retest/recheck/cli/subcommands/Diff.java

```diff
@@ -1,73 +1,169 @@
+			final Show c = new Show();
+			new CommandLine( c ).parseArgs( args );
+			c.run();
 			return;
 		}
+
+		final Path expectedSutStatePath = goldenMasterPath[0];
+		final Path actualSutStatePath = goldenMasterPath[1];
+
 		try {
-			final List<String> invalidFilters = FilterUtil.getInvalidFilters( exclude );
-			if ( !invalidFilters.isEmpty() ) {
-				FilterUtil.logWarningForInvalidFilters( invalidFilters );
-			} else {
-				printDiff();
+			if ( !hasValidExcludeOption() ) {
+				return;
 			}
+			final ActionReplayResult actionReplayResult =
+					createActionReplayResultFrom( expectedSutStatePath, actualSutStatePath );
+			final ActionReplayResult filteredResult = createFilteredActionReplayResult( actionReplayResult );
+			printGoldenMasterDifferences( filteredResult );
+			persistIfOutputOptionPresent( actionReplayResult );
 		} catch ( final Exception e ) {
 			ErrorHandler.handle( e );
 		}
 	}
 
-	private void printDiff() throws TestReportFormatException, IOException {
-		final TestReport report = TestReportUtil.load( testReport );
+	private boolean hasValidExcludeOption() {
+		final List<String> invalidFilters = FilterUtil.getInvalidFilters( exclude );
+		if ( !invalidFilters.isEmpty() ) {
+			FilterUtil.logWarningForInvalidFilters( invalidFilters );
+			return false;
+		}
+		return true;
+	}
+
+	private ActionReplayResult createActionReplayResultFrom( final Path expectedSutStatePath,
+			final Path actualSutStatePath ) {
+		final SutState expectedSutState = goldenMasterProvider
+				.loadGoldenMaster( createFile( expectedSutStatePath, RecheckProperties.DEFAULT_XML_FILE_NAME ) );
+		final SutState actualSutState = goldenMasterProvider
+				.loadGoldenMaster( createFile( actualSutStatePath, RecheckProperties.DEFAULT_XML_FILE_NAME ) );
+
+		return getActionReplayResult( expectedSutStatePath.toString(), expectedSutState, actualSutState );
+	}
+
+	private ActionReplayResult createFilteredActionReplayResult( final ActionReplayResult actionReplayResult ) {
 		final Filter excludeFilter = FilterUtil.getExcludeFilterFiles( exclude );
-		final Filter recheckIgnore = FilterUtil.loadRecheckIgnore();
-		final TestReportFilter filter = new TestReportFilter( new CompoundFilter( excludeFilter, recheckIgnore ) );
-		final TestReport filteredTestReport = filter.filter( report );
-		final TestReportPrinter printer = new TestReportPrinter( none() );
+		final TestReportFilter reportFilter =
+				new TestReportFilter( (excludeFilter) );
+		return reportFilter.filter( actionReplayResult );
+	}
+
+	private void persistIfOutputOptionPresent( final ActionReplayResult actionReplayResult ) throws IOException {
+		if ( directory != null ) {
+			persist( actionReplayResult, directory );
+			return;
+		}
+		logger.info( "\nThe displayed differences may be saved as test report by using the --output option." );
+	}
+
+	private File createFile( final Path path, final String defaultXmlFileName ) {
+		return new File( path.toString(), defaultXmlFileName );
```

Comment:
You could simply call, but if you do the method for the above, I would inline this method.
```suggestion
		return path.resolve(xml).toFile()
```

Reply:
I changed the creation of the file accordingly.